processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -94,20 +94,44 @@ public AggregateCombiner makeNullableAggregateCombiner()
   }
 
   /**
-   * Returns an AggregatorFactory that can be used to combine the output of aggregators from this factory.  This
-   * generally amounts to simply creating a new factory that is the same as the current except with its input
-   * column renamed to the same as the output column.
+   * Returns an AggregatorFactory that can be used to combine the output of aggregators from this factory. It is used
+   * when we know we have some values that were produced with this aggregator factory, and want to do some additional
+   * combining of them. This happens, for example, when merging query results from two different segments, or two
+   * different servers.
+   *
+   * The combining factory is often computed by simply creating a new factory that is the same as the current, except
+   * with its input column renamed to the same as the output column. For example, this aggregator:
+   *
+   * {@code {"type": "longSum", "fieldName": "foo", "name": "bar"}}
+   *
+   * Would become:
+   *
+   * {@code {"type": "longSum", "fieldName": "bar", "name": "bar"}}
+   *
+   * Somtimes, the type or other parameters of the aggregator will change. For example, a "count" aggregator's
+   * getCombiningFactory method will return a "longSum" aggregator, because counts are combined by summing.
+   *
+   * No matter what, `foo.getCombiningFactory()` and `foo.getCombiningFactory().getCombiningFactory()` should return
```

Comment:
Would be nice to test this automatically for all aggregator factories, created #7329.